lxd/api_1.0.go

```diff
@@ -240,7 +240,16 @@ func api10Put(d *Daemon, r *http.Request) Response {
 				return InternalError(err)
 			}
 
+			// Update the cached proxy function
 			d.updateProxy()
+
+			// Clear the simplestreams cache as it's tied to the old proxy config
+			imageStreamCacheLock.Lock()
```

Comment:
Nah, they'll just fail and try again if they happen to hit this.

What I want to make sure is that we flush it after updateProxy has run, since running it before would cause a much more likely race.